src/routes/transactions/helpers/imitation-transactions.helper.ts

```diff
@@ -0,0 +1,104 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { TransactionItem } from '@/routes/transactions/entities/transaction-item.entity';
+import {
+  isTransferTransactionInfo,
+  TransferDirection,
+} from '@/routes/transactions/entities/transfer-transaction-info.entity';
+import { isErc20Transfer } from '@/routes/transactions/entities/transfers/erc20-transfer.entity';
+import { Inject } from '@nestjs/common';
+
+export class ImitationTransactionsHelper {
+  private readonly vanityAddressChars: number;
+
+  constructor(
+    @Inject(IConfigurationService) configurationService: IConfigurationService,
+  ) {
+    this.vanityAddressChars = configurationService.getOrThrow(
+      'mappings.imitationTransactions.vanityAddressChars',
+    );
+  }
+
+  /**
+   * Filters out outgoing ERC20 transfers that imitate their direct predecessor in
+   * value and have a recipient address that is not the same but matches in vanity.
+   *
+   * @param transactions - list of transactions to filter
+   * @param previousTransaction - transaction to compare last {@link transactions} against
+   *
+   * Note: this only handles singular imitation transfers. It does not handle multiple
+   * imitation transfers in a row, nor does it compare batched multiSend transactions
+   * as the "distance" between those batched and their imitation may not be immediate.
+   */
+  filterOutgoingErc20ImitationTransfers(
+    transactions: Array<TransactionItem>,
+    previousTransaction: TransactionItem | undefined,
+  ): Array<TransactionItem> {
+    // TODO: Instead of filtering, mark transactions so client can display them differently
+    return transactions.filter((item, i, arr) => {
+      // Executed by Safe - cannot be imitation
+      if (item.transaction.executionInfo) {
+        return item;
+      }
```

Comment:
Nit: `filter` expects a boolean return value to decide whether to include the item in the resulting array. I think it's OK to return the item itself, but being stricter maybe return `true` is a bit more appropriate?

Reply:
Totally agree. Changed in 56f27fb54391f86585816b7df18a2d85265fc5ba.